analyze-project/README.md

```diff
@@ -0,0 +1,45 @@
+# `ni/python-actions/analyze-project`
+
+The `ni/python-actions/update-project-version` action analyzes the code quality
+of a Python project using various linters and type checkers including
+ni-python-styleguide, mypy (if the 'mypy' package is installed), and pyright
+(if the 'pyright' package is installed).uses Poetry to update the version of a Python
+project and creates a pull request to modify its `pyproject.toml` file. Publish workflows can use
+this to update the version in `pyproject.toml` for the next build.
+
+This action requires Poetry, so you must call `ni/python-actions/setup-python` and
+`ni/python-actions/setup-poetry` first.
+
+## Usage
+
+```yaml
+steps:
+- uses: ni/python-actions/setup-python@v0.2
+- uses: ni/python-actions/setup-poetry@v0.2
+- uses: ni/python-actions/analyze-project@v0.2
+```
+
+## Inputs
+
+### `project-directory`
+
+You can specify `project-directory` to indicate the location of the pyproject.toml
+file associated with the Python project you are analyzing.
+
+```yaml
+- uses: ni/python-actions/update-project-version@v0.2
+  with:
+    project-directory: ${{ github.workspace }}/packages/myproject
+```
+
+### `extras`
+
+If there are extras you need to install from your pyproject.toml, specify a space-separated list
+of extra groups to install. For example, 
+
+```yaml
+- uses: ni/python-actions/analyze-project@v0.2
+  with:
+    project-directory: ${{ github.workspace }}/packages/myproject
+    extras: 'docs drivers'
```

Comment:
This action didn't exist in v0.2, so let's use the floating v0 tag. 

I can't specify a suggestion here because the triple backticks confuses Github.